Algorithms/Search/BinarySearcher.cs

```diff
@@ -43,5 +44,53 @@ public int FindIndex(T[] sortedData, T item)
 
             return -1;
         }
+
+        /// <summary>
+        /// Finds index of item in collection that equals to item searched for,
+        /// time complexity: O(log(n)),
+        /// space complexity: O(1),
+        /// where n - collection size.
+        /// </summary>
+        /// <param name="collection">Sorted collection to search in.</param>
+        /// <param name="item">Item to search for.</param>
+        /// <returns>Index of item that equals to item searched for or -1 if none found.</returns>
+        public int FindIndexRecursive(IList<T>? collection, T item)
+        {
+            collection = collection ?? throw new ArgumentNullException(nameof(collection));
```

Comment:
Codacy found an issue: [Use the '??=' operator here.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6286772)

Reply:
A throw expression is not allowed in this context.